src/lib/IFrameCommunicator.ts

```diff
@@ -68,20 +54,57 @@ export class IFrameCommunicator {
   public sendMessage = <M extends keyof IFrameMessagesMap>(
     type: M,
     data: Parameters<IFrameMessagesMap[M]>[0],
-    metadata?: Parameters<IFrameMessagesMap[M]>[1]
+    metadata?: Parameters<IFrameMessagesMap[M]>[1],
+    operationId?: string
   ): void => {
     if (!isInsideIFrame()) {
       console.error("Warning: can't send an iframe message, because the application is not hosted in an iframe");
       return;
     }
 
-    const message: IFrameMessage<M> = { type, data, metadata };
+    const message: IFrameMessage<M> = { type, data, metadata, operationId };
     window.parent.postMessage(message, '*');
   };
 
+  public sendMessageWithResponse = <M extends keyof IFrameMessagesMap>(
+    type: M,
+    data: Parameters<IFrameMessagesMap[M]>[0],
+    callback: () => void,
+    metadata?: Parameters<IFrameMessagesMap[M]>[1]
+  ): void => {
+    const operationId = this.createUuid();
+    this.registerCallback(callback, operationId);
+    this.sendMessage(type, data, metadata, operationId);
+  };
+
   private onMessage = (event: MessageEvent): void => {
     if (!event.data) return;
     const message = event.data as IFrameMessage<any>;
     this.events.emit(message.type, message.data);
+
+    if (message.operationId) {
+      this.executeCallback(message.operationId, {});
+    }
   };
+
+  public createUuid(): string {
+    const randomUint32 = window.crypto.getRandomValues(new Uint32Array(1))[0];
+    return randomUint32.toString(16);
```

Comment:
I think we should move this function to a separate utils file because it is not related to the iframe communication and may be useful for other purposes in the future (e.g. generating unique IDs during rendering, etc.).